### ISSUE TITLE
no-duplicate-imports should ignore export *

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12760]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2015,
        sourceType: "module"
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWR1cGxpY2F0ZS1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IFwiaW5jbHVkZUV4cG9ydHNcIjogdHJ1ZSB9XSAqL1xuXG5pbXBvcnQgZm9vIGZyb20gJ2Zvbyc7XG5leHBvcnQgKiBmcm9tICdmb28nO1xuXG5pbXBvcnQgeyBhIH0gZnJvbSAnYmFyJztcbmV4cG9ydCAqIGZyb20gJ2Jhcic7XG5cbmltcG9ydCAqIGFzIG5zIGZyb20gJ2Jheic7XG5leHBvcnQgKiBmcm9tICdiYXonO1xuXG5leHBvcnQgeyBhIGFzIGIgfSBmcm9tICdxdXV4JztcbmV4cG9ydCAqIGZyb20gJ3F1dXgnO1xuXG5leHBvcnQgeyBkZWZhdWx0IH0gZnJvbSAncXV1dXgnO1xuZXhwb3J0ICogZnJvbSAncXV1dXgnOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

```js
/* eslint no-duplicate-imports: ["error", { "includeExports": true }] */

import foo from 'foo';
export * from 'foo';

import { a } from 'bar';
export * from 'bar';

import * as ns from 'baz';
export * from 'baz';

export { a as b } from 'quux';
export * from 'quux';

export { default } from 'quuux';
export * from 'quuux';
```

```bash
eslint index.js
```

**What did you expect to happen?**

I think no errors. The rule requires only 1 import/export statement per source module. I'm not sure how to rewrite any of these 5 examples to achieve that. Or any other code with  `export *` and another import/export from the same source, including the test case:

```js
{
    code: "import os from \"os\";\nexport * from \"os\";",
    options: [{ includeExports: true }],
    errors: [{ messageId: "exportAs", data: { module: "os" }, type: "ExportAllDeclaration" }]
}
```

 If I'm not missing something, this rule should probably ignore `export *`.

**What actually happened? Please include the actual, raw output from ESLint.**

Errors for all 5 `export *` declarations:

```
   4:1  error  'foo' export is duplicated as import  no-duplicate-imports
   7:1  error  'bar' export is duplicated as import  no-duplicate-imports
  10:1  error  'baz' export is duplicated as import  no-duplicate-imports
  13:1  error  'quux' export is duplicated           no-duplicate-imports
  16:1  error  'quuux' export is duplicated          no-duplicate-imports
```

**Are you willing to submit a pull request to fix this bug?**

Yes.

**Comment by kaicataldo** [comment]
Agreed!

**Comment by mdjermanovic** [comment]
I'm working on this
